src/components/DynamicDimension/DynamicDimension.js

```diff
@@ -1,12 +1,9 @@
-import React, { Component, Fragment } from 'react'
+import React, { Component } from 'react'
 import PropTypes from 'prop-types'
-import DialogTitle from '@material-ui/core/DialogTitle'
-import DialogContent from '@material-ui/core/DialogContent'
 
 import ItemSelector from '../ItemSelector/ItemSelector'
 import FilterField from '../FilterField'
 import { apiFetchItemsByDimension } from '../../api/dimensions'
-import { styles } from './styles/DynamicDimension.style'
 
 export class DynamicDimension extends Component {
```

Comment:
Do we convert all components into functional components or do we keep some classes where we have a more "complex" state?

Reply:
Tried to limit the scope somewhat and not convert the ones with complex state and hooks. Feels like this PR was big enough as it was already 😄